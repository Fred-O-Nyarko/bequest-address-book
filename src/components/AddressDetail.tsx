import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  List,
  ListItem,
  ListItemText,
  Typography,
} from "@mui/material";
import { IAddress, TModalID } from "../shared/types";

interface IAddressDetailProps {
  open: TModalID;
  setOpenModal: (m: TModalID) => void;
  addressDetail: IAddress | null;
}

const AddressDetail = ({
  open,
  setOpenModal,
  addressDetail,
}: IAddressDetailProps) => {
  return (
    <Dialog open={!!open} onClose={() => setOpenModal(null)} fullWidth>
      <DialogTitle>Address Detail</DialogTitle>
      <DialogContent dividers>
        <List>
          <ListItem>
            <ListItemText
              primary="Address Line 1"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineOne}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 2"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineTwo ?? 'N/A'}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Address Line 3"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.lineThree ?? 'N/A'}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Postcode"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.postCode}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Town/City"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.town}</Typography>}
            />
          </ListItem>
          <ListItem>
            <ListItemText
              primary="Country"
              primaryTypographyProps={{
                fontWeight: 600,
              }}
              secondary={<Typography>{addressDetail?.country}</Typography>}
            />
          </ListItem>
        </List>
      </DialogContent>
      <DialogActions>
        <Button onClick={() => setOpenModal(null)}>Cancel</Button>
      </DialogActions>
    </Dialog>
  );
};

export default AddressDetail;
