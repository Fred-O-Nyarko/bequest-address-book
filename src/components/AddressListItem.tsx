import { Delete } from "@mui/icons-material";
import {
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText,
  Typography,
  IconButton,
} from "@mui/material";
import { deepOrange } from "@mui/material/colors";
import { IAddressesResponse } from "../shared/types";

interface IAddressListItemProps {
    address: IAddressesResponse;
    onDelete: (address: IAddressesResponse) => void;
    postCode: string;
}
const AddressListItem = ({ onDelete, address, postCode }: IAddressListItemProps) => {
    
  return (
    <ListItem
      alignItems="flex-start"
      divider
      style={{
        backgroundColor: "#f6f5ea",
        borderRadius: "0.2rem"
      }}
      secondaryAction={
        <IconButton edge="end" aria-label="delete" onClick={() => onDelete(address)}>
          <Delete />
        </IconButton>
      }
    >
      <ListItemAvatar>
        <Avatar sx={{ bgcolor: deepOrange[500] }}>{postCode.slice(-2)}</Avatar>
      </ListItemAvatar>
      <ListItemText
        primary="Brunch this weekend?"
        primaryTypographyProps={{
            fontWeight: 600,
        }}
        secondary={
          <>
            <Typography
              sx={{ display: "inline" }}
              component="span"
              variant="body2"
              color="text.primary"
              fontWeight={600}
            >
              Ali Connors
            </Typography>
            {" — I'll be in your neighborhood doing errands this…"}
          </>
        }
      />
    </ListItem>
  );
};

export default AddressListItem;
