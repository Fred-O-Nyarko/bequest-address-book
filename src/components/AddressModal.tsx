import {
  Dialog,
  DialogTitle,
  DialogContent,
  DialogContentText,
  TextField,
  DialogActions,
  Button,
  Grid,
} from "@mui/material";
import { FormikState, FormikErrors } from "formik";
import { IAddressesResponse, IFormValues } from "../shared/types";
import SearchBox from "./SearchBox";

interface IAddressModalProps {
  open: boolean;
  setOpenModal: (m: boolean) => void;
  onSearch: any;
  options: IAddressesResponse[];
  loading: boolean;
  openSearch: boolean;
  setOpenSearch: (m: boolean) => void;
  getOptionLabel: (o: IAddressesResponse|string) => string;
  isOptionEqualToValue: (
    o: IAddressesResponse,
    v: IAddressesResponse
  ) => boolean;
  handleBlur: (e: React.FocusEvent<HTMLInputElement>) => void;
    onSubmit: () => void;
    handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
    setFieldValue: (name: string, value: any) => void;
    resetForm:  (nextState?: Partial<FormikState<IFormValues>> | undefined) => void;
    getError: (name: string) => string;
    errors: FormikErrors<IFormValues>;

}

const AddressModal = ({
  open,
  setOpenModal,
  onSearch,
  options,
  loading,
  openSearch,
  setOpenSearch,
  getOptionLabel,
  isOptionEqualToValue,
  handleBlur,
    onSubmit,
    handleChange,
    setFieldValue,
    resetForm,
    getError,
    errors,
    
}: IAddressModalProps) => {
  const handleClose = () => {
    setOpenModal(false);
  };



  const onAdd = () => {
    if (errors) return;
    resetForm();
    setOpenModal(false);
    onSubmit();
  };

  const onSearchChange = (value: string) => {
    handleChange(value as unknown as React.ChangeEvent<HTMLInputElement>);
    onSearch(value);
  };

  return (
    <Dialog open={open} onClose={handleClose}>
      <DialogTitle>Add Address</DialogTitle>
      <DialogContent>
        <DialogContentText>
          Fill in details below to add a new address.
        </DialogContentText>
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <TextField
              autoFocus
              name="lineOne"
              margin="dense"
              label="Address 1"
              fullWidth
              variant="standard"
              onChange={handleChange}
              onBlur={handleBlur}
              required
              error={!!getError("lineOne")}
              helperText={getError("lineOne")}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              autoFocus
              name="lineTwo"
              margin="dense"
              label="Address 2"
              fullWidth
              variant="standard"
              onChange={handleChange}
              onBlur={handleBlur}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              autoFocus
              name="lineThree"
              margin="dense"
              label="Address 3"
              fullWidth
              variant="standard"
              onChange={handleChange}
              onBlur={handleBlur}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              autoFocus
              name="town"
              margin="dense"
              label="Town"
              fullWidth
              variant="standard"
              onChange={handleChange}
              onBlur={handleBlur}
              required
              error={!!getError("town")}
              helperText={getError("town")}
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              autoFocus
              name="postCode"
              margin="dense"
              label="Postal Code"
              fullWidth
              variant="standard"
              onChange={handleChange}
              onBlur={handleBlur}
              required
              error={!!getError("postCode")}
              helperText={getError("postCode")}
            />
          </Grid>
          <Grid item xs={12}>
           
            <SearchBox
              searchFxn={onSearchChange}
              changeFxn={setFieldValue}
              options={options}
              label="Country"
              inputName="country"
              setFieldValue={setFieldValue}
              loading={loading}
              required
              getError={getError}
              open={openSearch}
              setOpen={setOpenSearch}
              getOptionLabel={getOptionLabel}
              isOptionEqualToValue={isOptionEqualToValue}
            />
          </Grid>
        </Grid>
      </DialogContent>
      <DialogActions>
        <Button onClick={() => setOpenModal(false)}>Cancel</Button>
        <Button onClick={onAdd}>Add</Button>
      </DialogActions>
    </Dialog>
  );
};

export default AddressModal;
