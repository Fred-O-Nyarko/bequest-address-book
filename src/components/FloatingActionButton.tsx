import { Box, Fab } from "@mui/material";
import AddIcon from "@mui/icons-material/Add";

interface IFloatingActionButtonProps {
  onClick: () => void;
}

const FloatingActionButton = ({ onClick }: IFloatingActionButtonProps) => {
  return (
    <Box position="absolute" bottom={32} right={0} marginRight={8}>
      <Fab
        variant="extended"
        onClick={onClick}
        color="primary"
        style={{
          backgroundColor: "#ff4c50",
        }}
      >
        <AddIcon sx={{ mr: 1 }} />
        New Address
      </Fab>
    </Box>
  );
};

export default FloatingActionButton;
