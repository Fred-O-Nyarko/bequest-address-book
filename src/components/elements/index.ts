export { default as SearchBox } from "./SearchBox";
export { default as EmptyState } from "./EmptyState";
export { default as Loader } from "./Loader";
export { default as FloatingActionButton } from "./FloatingActionButton";
export { default as Notification } from "./Notification";