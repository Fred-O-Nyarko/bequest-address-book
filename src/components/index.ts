export {default as SearchBox} from './SearchBox';
export {default as EmptyState} from './EmptyState'
export {default as Loader} from './Loader'
export {default as FloatingActionButton} from './FloatingActionButton'
export {default as AddressModal} from './AddressModal'
export {default as AddressListItem} from './AddressListItem'
export {default as Notification } from './Notification'