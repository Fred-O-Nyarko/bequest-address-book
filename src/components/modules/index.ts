export { default as AddressModule } from "./address/Address";
