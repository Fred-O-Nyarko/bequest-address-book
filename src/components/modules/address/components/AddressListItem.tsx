import { Delete } from "@mui/icons-material";
import {
  ListItem,
  ListItemAvatar,
  Avatar,
  ListItemText,
  Typography,
  IconButton,
  ListItemButton,
} from "@mui/material";
import { deepOrange } from "@mui/material/colors";
import { IAddress } from "src/shared";

interface IAddressListItemProps {
  address: IAddress;
  onDelete: (id: string) => void;
  onClick: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
}
const AddressListItem = ({
  onDelete,
  address,
  onClick,
}: IAddressListItemProps) => {
  const { postCode, lineOne, country, town, id } = address || {};
  return (
    <ListItem
      alignItems="flex-start"
      divider
      style={{
        backgroundColor: "#f6f5ea",
        borderRadius: "0.2rem",
      }}
      secondaryAction={
        <IconButton
          edge="end"
          aria-label="delete"
          onClick={() => onDelete(id)}
          id={`delete-btn-${postCode.split(" ").join("")}`}
        >
          <Delete />
        </IconButton>
      }
    >
      <ListItemButton role={undefined} onClick={onClick} dense>
        <ListItemAvatar>
          <Avatar sx={{ bgcolor: deepOrange[500] }}>
            {postCode.slice(0, 2)}
          </Avatar>
        </ListItemAvatar>
        <ListItemText
          primary={lineOne}
          primaryTypographyProps={{
            fontWeight: 600,
          }}
          secondary={
            <>
              <Typography
                sx={{ display: "inline" }}
                component="span"
                variant="body2"
                color="text.primary"
              >
                {town}, {country}
              </Typography>
            </>
          }
        />
      </ListItemButton>
    </ListItem>
  );
};

export default AddressListItem;
