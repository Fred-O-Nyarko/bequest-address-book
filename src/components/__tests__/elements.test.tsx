/* eslint-disable testing-library/prefer-screen-queries */
import { cleanup, render } from "@testing-library/react";
import { Provider } from "react-redux";
import { EmptyState, FloatingActionButton } from "src/components";
import { store } from "src/redux";

describe("components test for elements", () => {
  afterEach(cleanup);
  it("renders EmptyState component with empty state message", () => {
    const { getByTestId, container } = render(<EmptyState message="Testing" />);

    expect(getByTestId("empty-message")).toHaveTextContent("Testing");
    expect(container).toMatchSnapshot();
  });

  it("renders FloatingActionButton", () => {
    const { container } = render(
      <Provider store={store}>
        <FloatingActionButton />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
