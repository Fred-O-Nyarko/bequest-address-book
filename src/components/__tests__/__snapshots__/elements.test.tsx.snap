// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components test for elements renders EmptyState component 1`] = `
<div>
  <div
    class="flex flex-col items-center gap-8 w-full"
    style="display: flex; flex-direction: column; align-items: center; justify-content: center; gap: 2rem;"
  >
    <svg
      class="h-48 mt-10"
      style="height: 12rem; margin-top: 3rem;"
    >
      empty.svg
    </svg>
    <div
      class="text-center text-lg"
      data-testid="empty-message"
    >
      Testing
    </div>
  </div>
</div>
`;

exports[`components test for elements renders EmptyState component with empty state message 1`] = `
<div>
  <div
    class="flex flex-col items-center gap-8 w-full"
    style="display: flex; flex-direction: column; align-items: center; justify-content: center; gap: 2rem;"
  >
    <svg
      class="h-48 mt-10"
      style="height: 12rem; margin-top: 3rem;"
    >
      empty.svg
    </svg>
    <div
      class="text-center text-lg"
      data-testid="empty-message"
    >
      Testing
    </div>
  </div>
</div>
`;

exports[`components test for elements renders FloatingActionButton 1`] = `
<div>
  <div
    class="MuiBox-root css-y8gynq"
  >
    <button
      class="MuiButtonBase-root MuiFab-root MuiFab-extended MuiFab-sizeLarge MuiFab-primary css-nubenn-MuiButtonBase-root-MuiFab-root"
      style="background-color: rgb(255, 76, 80);"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        style="margin-right: 8px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
         New Address
      </p>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
